addons/common/Cfg3DEN.hpp

```diff
@@ -129,14 +129,127 @@ class Cfg3DEN {
+                    onKillFocus = QUOTE(\
+                        private _code = ctrlText (_this select 0);\
+                        (ctrlParent (_this select 0) getVariable [ARR_2('GVAR(InitAttributeValue_Object)', controlNull)]) ctrlSetText _code;\
+                        if (_code != '') then {\
+                            _code = 'call{' + _code + '}';\
+                        };\
+                        (ctrlParentControlsGroup (_this select 0) controlsGroupCtrl 100) ctrlSetText _code;\
+                    );
+                };
+            };
+        };
+
+        class GVAR(EditCodeMulti6_InitGroup): GVAR(EditCodeMulti6) {
+            class Controls: Controls {
+                class Background: Background {};
+                class Title: Title {};
+                class Value: Value {};
+                class GVAR(ValueInit): GVAR(Value) {
+                    // Copies contents of editable init box into the hidden
+                    // variant and into CBA_Init attribute.
+                    // Automatically adds call-block wrapper to enable
+                    // the usage of local variables and return values.
+                    onKillFocus = QUOTE(\
+                        private _code = ctrlText (_this select 0);\
+                        (ctrlParent (_this select 0) getVariable [ARR_2('GVAR(InitAttributeValue_Group)', controlNull)]) ctrlSetText _code;\
+                        if (_code != '') then {\
+                            _code = 'call{' + _code + '}';\
+                        };\
+                        (ctrlParentControlsGroup (_this select 0) controlsGroupCtrl 100) ctrlSetText _code;\
+                    );
+                };
+            };
+        };
+
+        class GVAR(EditCodeMulti6_InitLogic): GVAR(EditCodeMulti6) {
+            class Controls: Controls {
+                class Background: Background {};
+                class Title: Title {};
+                class Value: Value {};
+                class GVAR(ValueInit): GVAR(Value) {
+                    // Copies contents of editable init box into the hidden
+                    // variant and into CBA_Init attribute.
+                    // Automatically adds call-block wrapper to enable
+                    // the usage of local variables and return values.
+                    onKillFocus = QUOTE(\
+                        private _code = ctrlText (_this select 0);\
+                        (ctrlParent (_this select 0) getVariable [ARR_2('GVAR(InitAttributeValue_Logic)', controlNull)]) ctrlSetText _code;\
+                        if (_code != '') then {\
+                            _code = 'call{' + _code + '}';\
+                        };\
+                        (ctrlParentControlsGroup (_this select 0) controlsGroupCtrl 100) ctrlSetText _code;\
+                    );
+                };
+            };
+        };
+
+        class GVAR(InitAttribute_Object): GVAR(EditCodeMulti3) {
```

Comment:
You shouldn't need to use an edit control if this just a dummy that saves a copy of the init's value - can use an empty controls group.